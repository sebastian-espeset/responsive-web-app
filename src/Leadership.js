import {React, useState, useEffect} from 'react';
import axios from 'axios';
import { StyledLeadershipContainer, StyledLeadershipHeadline, StyledLeadershipLink, StyledLeadershipLinksContainer } from './styles/Leadership.style';
import  Leader  from './Leader.js';

//build fetch function 
const leadershipUrl='https://beamtech.github.io/responsive-web-app-kata/leadership-team.json'
export default function Leadership() {
  const [leaders,setLeaders]=useState([])
  useEffect(() => {
    axios.get(leadershipUrl)
        .then(res => {
          setLeaders(res.data)
        })
        .catch(err=>console.log(err))
},[]);
  return (
    <StyledLeadershipContainer>
      <StyledLeadershipHeadline data-testid='leadership-headline-test-id'>
        Our Leadership Team 
      </StyledLeadershipHeadline>
      <div>────────</div>
      <StyledLeadershipLinksContainer>
        {leaders.length>1 ? leaders.map((leader)=>{
        return <Leader img={leader.image} name={leader.name} title={leader.title} url={leader.title}/>
        }):<div>Leaders loading</div>}
      </StyledLeadershipLinksContainer>
    </StyledLeadershipContainer>
  )
}
