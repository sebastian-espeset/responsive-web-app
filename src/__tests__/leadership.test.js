import {render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Leadership from '../Leadership';

afterEach(()=>{
    cleanup();
});

test('Leaderships section headline should render',()=>{
    //Arrange
    render(<Leadership/>)
    //Act
    const leadershipHeadline = screen.getByTestId('leadership-headline-test-id');
    //Assert
    expect(leadershipHeadline).toBeInTheDocument();
});

// test will create a 'snapshot' of the html file, and then compare it to potential changes made    
test('leadership component to match snapshot',()=>{
    const tree = renderer.create(<Leadership/>).toJSON();
    expect(tree).toMatchSnapshot();
})