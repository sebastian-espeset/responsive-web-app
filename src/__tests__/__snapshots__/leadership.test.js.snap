// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`leadership component to match snapshot 1`] = `
<div
  className="sc-bczRLJ cBWjmY"
>
  <h1
    className="sc-gsnTZi itvaoF"
    data-testid="leadership-headline-test-id"
  >
    Our Leadership Team
  </h1>
  <div>
    ────────
  </div>
  <div
    className="sc-dkzDqf BCGrc"
  >
    <div>
      Leaders loading
    </div>
  </div>
</div>
`;
